### PR TITLE
Let's you talk through action figures, plushies, and toy mechs with `.l` and `.r`. Also a big clean up of `say` because its support for non-mobs was lackluster.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- You can use `.l` or `.r` to talk via a plushie, toy mech, or action figure, as you would with a ventriloquist dummy. 

image

- Speaking via an object like this cannot be transmitted over open mic radio.

- You can now pass in message mods into `say` directly. 
   - This allows for users to force a say to use certain configurations, such as a certain radio key, whisper mode, singing, etc. without necessitating that you manually add in the relevant key. 
   - For example, instead of doing `say("#[message]")`, you can just do `say(message, message_mods = list(WHISPER_MODE = MODE_WHISPER))`. This may be handy in a few places, and lets us be a bit more "official" than manually doing stuff like `say(";GRAGHHG")`. 

- Doc'd a lot of say related procs. I can't believe they weren't doc'd before. 

- Made `/atom/movable` level respect `can_speak`. 

## Why It's Good For The Game

I saw someone "talk via" a plushie like a kid pretending to be one of their toys and I thought it was funny so I thought I should make it a thing. 

But I didn't want it to transmit over radio because that wouldn't make sense. So I had to clean up a lot of code because say is a mess. 

## Changelog

:cl: Melbert
add: You can talk via plushies, action figures, and toy mechs via `.l` or `.r`, same as a ventriloquist dummy or decapitated dead
del: Talking via a plushie, action figure, toy mech, ventriloqiest dummy, decapitated head, or marionette won't transmit over open mic comms
refactor: Some say refactor and code cleanup. Vending machines now properly respect being shut up. Report any oddities. 
/:cl:

### CODE REVIEW

File: code/__DEFINES/say.dm [hunk 1]
```diff
@@ -75,8 +78,11 @@
 #define SPAN_HELIUM "small"
 
 //bitflag #defines for return value of the radio() proc.
+/// Makes the use italics
```
Comment (san7890) [inline code/__DEFINES/say.dm:81]:
```suggestion
/// Makes the message use italics
```

sentence needs a subject

File: code/datums/elements/toy_talk.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/datum/element/toy_talk/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ITEM_TALK_INTO, PROC_REF(do_talk))
+
+/datum/element/toy_talk/Detach(datum/source, ...)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_TALK_INTO)
```
Comment (san7890) [inline code/datums/elements/toy_talk.dm:17]: it's safe to invoke parent then unregister, right? the pattern is typically `UnregisterSignal(x); return ..()` so i just wanna make sure

File: code/game/objects/items/toy_mechs.dm [hunk 3]
```diff
@@ -263,12 +264,8 @@
...
 		. += span_notice("This toy has [wins] wins, and [losses] losses.")
 
-/**
- * Override the say proc if they're mute
- */
-/obj/item/toy/mecha/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null, message_range = 7, datum/saymode/saymode = null)
-	if(!quiet)
-		. = ..()
+/obj/item/toy/mecha/can_speak(allow_mimes)
+	return ..() && !quiet
```
Comment (san7890) [inline code/game/objects/items/toy_mechs.dm:268]:
```suggestion
	return !quiet && ..()
```

shortcircuiting so we don't have to waste time on a useless parent call (unless parent _needs_ to be invoked for some reason)

File: code/modules/mob/living/basic/space_fauna/statue/statue.dm [hunk 4]
```diff
@@ -55,6 +55,7 @@
 /mob/living/basic/statue/Initialize(mapload)
 	. = ..()
 	ADD_TRAIT(src, TRAIT_UNOBSERVANT, INNATE_TRAIT)
+	ADD_TRAIT(src, TRAIT_MUTE, INNATE_TRAIT)
```
Comment (san7890) [inline code/modules/mob/living/basic/space_fauna/statue/statue.dm:58]:
```suggestion
	add_traits(list(TRAIT_MUTE, TRAIT_UNOBSERVANT), INNATE_TRAIT)
```

File: code/modules/mob/living/brain/brain_say.dm [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
+	datum/language/language,
+	ignore_spam = FALSE,
+	forced,
+	filterproof = FALSE,
+	message_range = 7,
+	datum/saymode/saymode,
+	list/message_mods = list(),
+)
+	if(prob(emp_damage * 4))
+		if(prob(10))//10% chane to drop the message entirely
```
Comment (san7890) [inline code/modules/mob/living/brain/brain_say.dm:15]:
```suggestion
		if(prob(10)) //10% chance to drop the message entirely
```

may as well fix it (at least the typo) while we're futzing with the diff

File: code/modules/mob/living/brain/brain_say.dm [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+			return
+		message = Gibberish(message, emp_damage >= 12)//scrambles the message, gets worse when emp_damage is higher
+
+	return ..()
 
-		..()
+/mob/living/brain/can_speak(allow_mimes)
+	return ..() && istype(container, /obj/item/mmi)
```
Comment (san7890) [inline code/modules/mob/living/brain/brain_say.dm:22]: i think we should also flip the parent call and the `istype()` here as well for short-circuiting.

File: code/modules/vending/_vending.dm [hunk 7]
```diff
@@ -276,8 +276,8 @@
 	QDEL_NULL(sec_radio)
 	return ..()
 
-/obj/machinery/vending/can_speak()
-	return !shut_up
+/obj/machinery/vending/can_speak(allow_mimes)
+	return ..() && is_operational && !shut_up
```
Comment (san7890) [inline code/modules/vending/_vending.dm:280]: ditto short-circuiting

**Review by san7890 — CHANGES_REQUESTED** [review]
nits but also some code considerations that would improve everything a bit

File: code/datums/elements/toy_talk.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/elements/toy_talk.dm:17]: Yes, element Detach parent call doesn't do anything important

File: code/modules/mob/living/brain/brain_say.dm (hunk 6 above)
Comment (MrMelbert) [reply code/modules/mob/living/brain/brain_say.dm:22]:
```suggestion
	return istype(container, /obj/item/mmi) && ..()
```

File: code/modules/vending/_vending.dm (hunk 7 above)
Comment (MrMelbert) [reply code/modules/vending/_vending.dm:280]:
```suggestion
	return is_operational && !shut_up && ..()
```

**Comment by nukularpower** [comment]
so does this fix the mime's ventriloquist dummy?

**Comment by MrMelbert** [comment]
no

**Pull request merged**